content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+2015-11-17
+:jbake-type: page
+:toc: macro
+
+ifdef::env-github,env-browser[:outfilesuffix: .adoc]
+
+toc::[]
+
+== Introduction
+
+Reader conditionals are a feature added in Clojure 1.7. They are designed to allow different dialects of Clojure to share common code that is mostly platform independent, but contains some platform dependent code. If you are writing code across multiple platforms that is mostly independent you should use `.clj` and `.cljs` for this.
+
+Reader conditionals are integrated into the Clojure Compiler, and don't require any extra tooling beyond Clojure 1.7 or greater. To use Reader conditionals, all you need is for your file to have a `.cljc` extension and to use Clojure 1.7 or ClojureScript 0.0-3196 or higher. Reader conditionals are data, and can be manipulated like ordinary Clojure data. For more technical details, see the reference page on <<xref/../../reference/reader#,the reader>>.
+
+There are two types of reader conditionals, standard and splicing. The standard conditional reader behaves similarly to a traditional `cond`. The syntax for usage is `#?` and looks like:
+
+[source,clojure]
+----
+#?(:clj  (Clojure expression)
+   :cljs (ClojureScript expression)
+   :clr  (Clojure CLR expression))
+----
+
+The syntax for a splicing conditional read is `#?@`. It is used to splice lists into the containing form. So the Clojure reader would read this:
+
+[source,clojure]
+----
+(defn build-list []
+  (list #?@(:clj  [5 6 7 8]
+            :cljs [1 2 3 4])))
+----
+
+as this:
+
+[source,clojure]
+----
+(defn build-list []
+  (list 5 6 7 8))
+----
+
+One important thing to note is that in Clojure 1.7 a splicing conditional reader cannot be used to splice in multiple top level forms. In concrete terms, this means you can't do this:
+
+[source,clojure]
+----
+;; Don't do this!, will throw an error
+#?@(:clj
+    [(defn clj-fn1 [] :abc)
+     (defn clj-fn2 [] :cde)])
+;; CompilerException java.lang.RuntimeException: Reader conditional splicing not allowed at the top level.
+----
+
+Instead you'd need to do wrap each function individually:
+
+[source,clojure]
+----
+#?(:clj (defn clj-fn1 [] :abc))
+#?(:clj (defn clj-fn2 [] :cde))
+----
+
+or use a `do` to wrap all of the top level functions:
+
+[source,clojure]
+----
+#?(:clj
+    (do (defn clj-fn1 [] :abc)
+        (defn clj-fn2 [] :cde)))
+----
+
+Which one you choose would probably depend on aesthetics, and what your plans for porting these functions to more Clojure platforms were. Let's go through some examples of places you might want to use these new reader conditionals.
+
+== Host interop
+
+Host interop is one of the biggest pain points that cljx and reader conditionals both solve. You may have a Clojure file that is almost pure Clojure, but needs to call out to the host environment for one function. https://github.com/lymingtonprecision/route-ccrs/blob/c579aea05504736f2cfbd31c3c755f7e25fdad77/src/route_ccrs/manufacturing_methods.cljc#L8-L10[This] is a classic example:
```

Comment:
`cljx` hasn't been introduced yet at this point. Perhaps restrict discussion to the historical appendix?